File: solutions/python/anagram/anagram.py
import collections


def find_anagrams(word: str, candidates: list[str]) -> list[str]:
    word_count = collections.Counter(word.lower())
    return [
        c for c in candidates
        if c.lower() != word.lower() and word_count == collections.Counter(c.lower())
    ]

File: solutions/ruby/anagram/anagram.rb
class Anagram
  private

  attr_accessor :word, :word_count

  def initialize(word)
    self.word = word.downcase
    self.word_count = count(word)
  end

  def count(word)
    word.downcase.chars.tally
  end

  public

  def match(candidates)
    candidates.select { |candidate| candidate.downcase != word && count(candidate) == word_count }
  end
end
